### ISSUE TITLE
Simulation: Unlock editing is required

### ISSUE THREAD

**Issue by dnsmichi** [Icinga/icingaweb2-module-businessprocess#226]
## Expected Behavior

I want to simulate a BP as easy as possible.

## Current Behavior

Since there is no documentation about it, I was looking into every option available. After a while I figured out that I needed to actually edit the BP in order to get the simulation.

## Possible Solution

Imho editing a BP is a different context than just simulating it. This also is true for the permission model, one that wants to simulate simple things doesn't necessarily need full admin rights to modify and edit the BP configuration.

The interface should be made more clear, "unlock editing" should clearly indicate that this is about changing the configuration, and the simulation is either found in there, or made a separate context.

My workflow for BPs defines as:

- Create the root node and the processes
- Edit the conditions, store and save as proposed by the interface
- Look at the BP and now I want to simulate certain conditions
- I don't know that I have to edit the BP again for this feature, it should just be there (granted that I need the permission to have it, normal users shouldn't have it)

![Screen Shot 2019-04-10 at 11 40 35](https://user-images.githubusercontent.com/382049/55869671-40f4ac00-5b87-11e9-8b1d-cb3cdaa2a430.png)
![Screen Shot 2019-04-10 at 11 40 47](https://user-images.githubusercontent.com/382049/55869672-40f4ac00-5b87-11e9-826f-f959ad5e15cb.png)

## Context

Writing a blog post and starting to work with the interface again. Playing the unexperienced user role here.

## Your Environment

* Module version (System - About): 2.2.0
* Icinga Web 2 version and modules (System - About): git master, a487f83f99e440bc71e268725a31943d9e29ea75
* Icinga 2 version (`icinga2 --version`): v2.10.2-257-g856d3a1
* Operating System and version: CentOS 7
* Webserver, PHP versions: httpd 2.4, PHP 7.1

**Comment by dnsmichi** [comment]
To make this more complex: Once you've changed a state in the simulation, you are asked to dismiss it. At this time you've left the editing context already, clicking 'dismiss' puts you back onto the locked view. I would expect to stay in the simulation mode here (which is hidden in the edit unlock context).

**Comment by nilmerg** [comment]
True, hiding this behind a permission is rather sub-optimal.

**Comment by theFeu** [comment]
Tbh, I feel like the UX is pretty far from optimal here anyway, like:
- too many icons that are unclear (a magic wand for simulation?)
- actions that don't need to be immediately visible (like the delete on the node itself, while everywhere else in the web the delete is in a form under editing)
- things being hidden behind way too many clicks with the unlock editing and similar
- The switch between tile and tree view icons being used on a page multiple times, while doing different things on click (one opens new tab with the selected, another one switching locally)

I totally agree with you and love your suggestion here, but I feel like we should generally rework the user interaction in the module rather than performing surgery on each individual part.

**Comment by dnsmichi** [comment]
Agreed. I had created this issue while working on a blog post for new BP features. I don't regularly use the module, so my "dumb user" card is played here. I'm fine with reworking the whole UX.
